### PR TITLE
Migrate to become a Server SDK

### PR DESCRIPTION
**Pull request by chrisbarbour**
* Client Updated to remove Target
* Caching implementation updated to get all Feature Configs and Segments.
* Auth implementation updated to authenticate with Server API Key.
* Evaluation Logic implemented by combining the information from the wiki and comparing it against the dot-net sdk.
* Polling implementation
* Metrics Publishing

### CODE REVIEW

**Review by bmjen — APPROVED** [review]
Smoke tested this manually and things look good. There is some more testing I'd like to do with Percentage Rollouts, but will do that separately and track necessary changes in JIRA.

File: force-app/main/classes/FFAuthServiceCached.cls [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            throw new AuthException('JWT token not valid! parts: ' + parts.size());
+        }
+        String decoded = EncodingUtil.base64Decode(parts.get(1)).toString();
+        return (FFModels.JWTClaims)JSON.deserialize(decoded, FFModels.JWTClaims.class);
+    }
+
+    private FFModels.AuthInfo reauthenticate(Integer retry) {
+        FFModels.AuthenticationRequest request = new FFModels.AuthenticationRequest();
+        request.apiKey = this.sdkKey;
+        request.target = new Map<String,String>{ 'identifier' => '' };
```
Comment (NarangAshish) [inline force-app/main/classes/FFAuthServiceCached.cls:57]: @chrisbarbour @bmjen  Is identifier suppose to be empty string here ?

Comment (chrisbarbour) [reply force-app/main/classes/FFAuthServiceCached.cls:57]: It is, the api call requires it to be sent even though it is unused, this allows a server sdk key to sent

Comment (NarangAshish) [reply force-app/main/classes/FFAuthServiceCached.cls:57]: Got this error "invalid key or target provided" if cache is empty.

Comment (chrisbarbour) [reply force-app/main/classes/FFAuthServiceCached.cls:57]: Please ensure that you are using a server sdk key as opposed to a client sdk key

Comment (NarangAshish) [reply force-app/main/classes/FFAuthServiceCached.cls:57]: valid point, thanks....:)

File: force-app/main/classes/FFPoller.cls [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            return instance.call('', new Map<String, Object>{ 'default' => args });
+        }
+
+        private void scheduleAgain() {
+            try {
+                String nextFireCron = calculateNextFireCron(iteration, configuration);
+    
+                iteration++;
+                FFPoller scheduledPoll = new FFPoller(configuration, iteration);
+                System.schedule('Polling - ' + nextFireCron, nextFireCron, scheduledPoll);
```
Comment (NarangAshish) [inline force-app/main/classes/FFPoller.cls:129]: Can we parameterize job name. 
In our use case, We have to find if Poller is already running for a tenant, do not start it again, and if for some reason it get killed, then start this again from different schedule job.

**Pull request merged**